tgui/packages/tgui/interfaces/ZoldorfPlayerShop/type.ts

```diff
@@ -7,24 +7,28 @@
 
 import { BooleanLike } from 'common/react';
 
-export interface ZoldorfPlayerShopData {
-  soul_products: ZoldorfSoulProductData[];
-  credit_products: ZoldorfCreditProductData[];
-  credits: number;
-  user_soul: number;
-}
-
-export interface ZoldorfProductData {
+export interface ZoldorfCommonProductData {
   name: string;
   stock: number;
   infinite: BooleanLike;
   img: string;
 }
 
-export interface ZoldorfSoulProductData extends ZoldorfProductData {
+export interface ZoldorfSoulProductData extends ZoldorfCommonProductData {
   soul_percentage: number;
 }
 
-export interface ZoldorfCreditProductData extends ZoldorfProductData {
+export interface ZoldorfCreditProductData extends ZoldorfCommonProductData {
   price: number;
 }
+
+export type ZoldorfProductData = ZoldorfSoulProductData | ZoldorfCreditProductData;
+
+export const isSoulProductData = (value: ZoldorfProductData): value is ZoldorfSoulProductData =>
```

Comment:
Typeguard, used to determine whether a non-specific one is one of the specific ones or not. TS can use this to do correct typing if code logic can only go one way, e.g. an `if` statement or a ternary.

Reply:
Thank you for explaining how to tackle it this way.